python/ray/data/read_api.py

```diff
@@ -320,6 +323,54 @@ def read_datasource(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_mongo(
+    uri: str,
+    database: str,
+    collection: str,
+    pipelines: List[List[Dict]],
```

Comment:
Why is this called pipelines instead of queries?

Reply:
It's a more general form of queries, for example, for the user that motivated this PR, their query is actually a pipeline of queries like:
- $match with a predicate
- $sort above results by a field
- $limit the number to return
- $project fields
Each of these steps is a mongo query, so it's kind of their way to compose more complex queries.

If there is a single step/query, then the inner List can just have one element.

Reply:
add as a comment in example? would be great to help explain

Reply:
Expanded the comment section and added and example pipeline.